### ISSUE TITLE
Generics disable type checking for concrete types inside interface

### ISSUE THREAD

**Issue by kvet** [microsoft/TypeScript#13680]
**TypeScript Version:**  2.1.5

**Code**

```ts
interface B {
    <T>(arg: C): T;
}

interface C {
    c: () => string;
}

let b: B = (e) => e.d()
```

**Expected behavior:**
Compiler should emit error that method `d` is not present in `C`.

**Actual behavior:**
Compilation finished without errors.

**Additional info**

My real scenario is quite complex. It is not working too. 

```ts
interface B {
    <T>(arg: C<T>): T;
}

interface C<T> {
    c: () => T;
}

let b: B = (e) => e.d()
```

**Comment by mhegazy** [comment]
contextual type does not apply in this case, since the signature is generic. The issue is the parameters in the source signature reference a type that does not exist in the target signature.

**Comment by RyanCavanaugh** [comment]
This errors now
